### PR TITLE
feat: allow controlling when a fixture is rendered

### PR DESCRIPTION
**Pull request by alestiago**
# Description

Changes `_renderFixtures` to `renderFixture`. This keeps the consistency with the other public render methods in `BodyComponent`.

Introduces new functionalities:
- Allow listening when a given fixture is rendered 
- Allows modifying the rendering logic for a given (one or more) fixtures. 

For example, avoiding a fixture to be rendered:
```dart
class MyBodyComponent extends BodyComponent {
...
void renderFixture(Canvas canvas, Fixture fixture) {
 final avoidRendering = condition; // Any condition here.
  if (avoidRendering) return;
  super.renderFixture(canvas, fixture);
}
...
}
```

## Checklist

- [X] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [X] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [X] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples`.

## Breaking Change

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

## Related Issues

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

**Comment by spydon** [comment]
I don't think we should rely on userdata for this (preferably we shouldn't rely on userdata for anything, it is a way too fragile pattern).

**Comment by alestiago** [comment]
(cc: @spydon )

This PR is an initial proposal to allow rendering only some Fixtures. Once settled, I'll update tests, examples and docs.

**Comment by alestiago** [comment]
> I don't think we should rely on userdata for this (preferably we shouldn't rely on userdata for anything, it is a way too fragile pattern).

Thanks @spydon for having a look. One doesn't need to rely on `userData`, as pointed out in the description, one can store those fixtures that would be skipped during rendering.

File: packages/flame_forge2d/lib/body_component.dart [hunk 1]
```diff
@@ -74,36 +74,52 @@ abstract class BodyComponent<T extends Forge2DGame> extends Component
...
+  /// to customize how fixtures are rendered. For example, you can filter out
+  /// fixtures that you don't want to render.
+  ///
+  /// **NOTE**: If [renderBody] is false, no fixtures will be rendered. Hence,
+  /// [renderFixture] is not called when [render]ing.
+  void renderFixture(
+    Canvas canvas,
+    Fixture fixture,
+  ) =>
+      _renderFixture(canvas, fixture);
```
Comment (st-pasha) [inline packages/flame_forge2d/lib/body_component.dart:121]: Looks like the `_renderFixture()` method can be inlined here, merging the two.

Comment (alestiago) [reply packages/flame_forge2d/lib/body_component.dart:121]: Indeed, that's how I had it initially. I think it fits well.

However, I went with `_renderFixture()` to keep with the file consistency, since `renderChain` has `_renderChain` and so forth. Those methods could be also a single method.

I'm in favour of not using `_renderFixture`. 

Wdyt @spydon ?

Comment (spydon) [reply packages/flame_forge2d/lib/body_component.dart:121]: Yeah, it doesn't add anything.

**Review by spydon — APPROVED** [review]
Great with some goldens here too :)

File: packages/flame_forge2d/pubspec.yaml [hunk 2]
```diff
@@ -4,17 +4,20 @@ version: 0.11.0
 homepage: https://github.com/flame-engine/flame/tree/main/packages/flame_forge2d
 
 environment:
-  sdk: ">=2.16.0 <3.0.0"
+  sdk: ">=2.17.0 <3.0.0"
```
Comment (luanpotter) [inline packages/flame_forge2d/pubspec.yaml:7]: should we keep this consistent across projects (ie 16)?
if we want to update we should make a separate PR updating every package

**Review by luanpotter — APPROVED** [review]
LGTM with one comment. Nice tests!

File: packages/flame_forge2d/pubspec.yaml (hunk 2 above)
Comment (luanpotter) [inline packages/flame_forge2d/pubspec.yaml:7]:
```suggestion
  sdk: ">=2.16.0 <3.0.0"
```

Comment (alestiago) [reply packages/flame_forge2d/pubspec.yaml:7]: For sure! When do we plan to update the SDK? Is there something stopping us from it?

Comment (spydon) [reply packages/flame_forge2d/pubspec.yaml:7]: I'm going to update after the release

Comment (spydon) [reply packages/flame_forge2d/pubspec.yaml]: And it is consistent on all other packages, if that was a question :)
